R/as-data-frame.R

```diff
@@ -73,3 +104,96 @@ as.data.frame.CrunchDataFrame <- function (x, row.names = NULL, optional = FALSE
     names(out) <- names(x)
     return(structure(out, class="data.frame", row.names=c(NA, -nrow(x))))
 }
+
```

Comment:
Absolutely, I was planning on looking into making the current promises closures and having the `$`, `[`, and `[[` methods evaluate them on the fly, but I think for the purposes of keeping this PR simply huge rather than massive, I'm going to defer those if possible.